src/Rules/TooWideTypehints/TooWideTypeCheck.php

```diff
@@ -176,6 +176,15 @@ public function checkFunctionReturnType(
 		if (count($returnStatements) === 0) {
 			return [];
 		}
+		if (
+			count($returnStatements) === 1
+			&& (
+				$nativeFunctionReturnType->isBoolean()->yes()
+				|| $phpDocFunctionReturnType->isBoolean()->yes()
```

Comment:
I thought we'd do this only for literal `return true` and `return false`, not for `return $this->foo->doFoo()`.

Reply:
good catch. fixed